tests/e2e/tests/pilot/routing_to_external_service_test.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pilot
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+)
+
+func TestExternalServiceRouteHeaders(t *testing.T) {
```

Comment:
TestExternalServiceAppendHeaders

Reply:
My plan is test that routing using `VirtualServices` work as expected when combined with `ExternalService`s.
httpbin.org is vital because it is external to the mesh and is a nice tool to examine the headers.
The tests for v1alpha1 are also dependent on httpbin.org. If we can't use httpbin.org, is there a different way to test my plan?